community/io/src/main/java/org/neo4j/io/pagecache/PageCache.java

```diff
@@ -61,6 +59,24 @@
      */
     PagedFile map( File file, int pageSize, OpenOption... openOptions ) throws IOException;
 
+    /**
+     * Ask for an already mapped paged file, backed by this page cache.
+     * <p>
+     * If mapping exist, the returned {@link Optional} will report {@link Optional#isPresent()} true and
+     * {@link Optional#get()} will return the same {@link PagedFile} instance that was initially returned my
+     * {@link #map(File, int, OpenOption...)}.
+     * If no mapping exist for this file, then returned {@link Optional} will report {@link Optional#isPresent()}
+     * false.
+     * <p>
+     * NOTE! User is responsible for closing the returned paged file.
+     *
+     * @param file The file to try to get the mapped paged file for.
+     * @return {@link Optional} containing the {@link PagedFile} mapped by this {@link PageCache} for given file, or an
+     * empty {@link Optional} if no mapping exist.
+     * @throws IOException if page cache has been closed or page eviction problems occur.
+     */
+    Optional<PagedFile> tryMappedPagedFile( File file ) throws IOException;
```

Comment:
I suggested the `try` prefix to hint at the optional return type. Another option could be `getExistingMapping`.